src/Workspaces/Core/Portable/Execution/AssetBuilder.cs

```diff
@@ -182,13 +187,20 @@ public AssetOnlyTreeNode(Solution solution)
 
             public Serializer Serializer { get; }
 
-            public Task<TResult> GetOrCreateAssetAsync<TKey, TValue, TResult>(
-                TKey key, TValue value, string kind, Func<TValue, string, CancellationToken, Task<TResult>> valueGetterAsync, CancellationToken cancellationToken)
+            public Asset GetOrCreateAsset<TKey, TValue>(
+                TKey key, TValue value, string kind, Func<TValue, string, CancellationToken, Asset> valueGetter, CancellationToken cancellationToken)
                 where TKey : class
-                where TResult : Asset
             {
                 Contract.ThrowIfNull(key);
-                return valueGetterAsync(value, kind, cancellationToken);
+                return valueGetter(value, kind, cancellationToken);
+            }
+
+            public TResult GetOrCreateChecksumObjectWithChildren<TKey, TValue, TResult>(
+                TKey key, TValue value, string kind, Func<TKey, TValue, string, CancellationToken, TResult> valueGetter, CancellationToken cancellationToken)
+                where TKey : class
+                where TResult : ChecksumObjectWithChildren
+            {
+                return Contract.FailWithReturn<TResult>("shouldn't be called");
```

Comment:
if it's private... why does it need to implement an interface?  Why can't AssetBuilder just hold onto it directly and call the functionality it needs?  Then you don't need to implement an interface method that you don't do anything with.